File: catalog/backends/java-quarkus/patterns/SparkEtlJob.java
package com.example.service;

import org.apache.spark.sql.*;
import static org.apache.spark.sql.functions.*;
import java.util.Properties;

// Pattern: DJM — Spark ETL job reading from project database
// Adapt: replace table names and transformations with domain logic
public class SparkEtlJob {

    public static void main(String[] args) {
        SparkSession spark = SparkSession.builder().appName("demo-etl").getOrCreate();
        Properties props = new Properties();
        props.put("user", "demo");
        props.put("password", "demo");
        props.put("driver", "org.postgresql.Driver");

        String url = "jdbc:postgresql://postgresql:5432/demo";
        Dataset<Row> df = spark.read().jdbc(url, "your_table", props);
        Dataset<Row> summary = df.groupBy("category")
            .agg(count("*").alias("total"), sum("amount").alias("revenue"));
        summary.write().mode(SaveMode.Overwrite).jdbc(url, "your_table_summary", props);
        spark.stop();
    }
}

File: catalog/backends/python-django/patterns/spark_etl_job.py
from pyspark.sql import SparkSession
from pyspark.sql.functions import count, sum as _sum

# Pattern: DJM — Spark ETL job reading from project database
# Adapt: replace table names and transformations with domain logic
DB_URL = "jdbc:postgresql://postgresql:5432/demo"
DB_PROPS = {"user": "demo", "password": "demo", "driver": "org.postgresql.Driver"}

spark = SparkSession.builder.appName("demo-etl").getOrCreate()

df = spark.read.jdbc(DB_URL, "your_table", properties=DB_PROPS)
summary = df.groupBy("category").agg(
    count("*").alias("total"),
    _sum("amount").alias("revenue"),
)
summary.write.jdbc(DB_URL, "your_table_summary", mode="overwrite", properties=DB_PROPS)

spark.stop()
